resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -71,7 +72,11 @@ abstract class DnsQueryContext implements FutureListener<AddressedEnvelope<DnsRe
         // Ensure we remove the id from the QueryContextManager once the query completes.
         promise.addListener(this);
 
-        if (parent.isOptResourceEnabled()) {
+        if (parent.isOptResourceEnabled() &&
+                // Only add the extra OPT record if there is not already one. This is required as only one is allowed
+                // as per RFC:
+                //  - https://datatracker.ietf.org/doc/html/rfc6891#section-6.1.1
+                !hasOptRecord(additionals)) {
```

Comment:
```suggestion
        // Only add the extra OPT record if there is not already one. This is required as only one is allowed
        // as per RFC: https://datatracker.ietf.org/doc/html/rfc6891#section-6.1.1
        if (parent.isOptResourceEnabled() && !hasOptRecord(additionals)) {
```

This looks cleaner I guess.

Reply:
I did it as I did as the comment is really only about the second branch.

Reply:
Oh, alright. :)